tools/topology/sof-tgl-rt711-rt1308.m4

```diff
@@ -189,18 +225,21 @@ DAI_CONFIG(ALH, 2, 0, SDW0-Playback,
 DAI_CONFIG(ALH, 3, 1, SDW0-Capture,
 	ALH_CONFIG(ALH_CONFIG_DATA(ALH, 3, 48000, 2)))
 
+ifdef(`AMPS',
+`
 #ALH SDW1 Pin2 (ID: 2)
 DAI_CONFIG(ALH, 0x102, 2, SDW1-Playback,
 	ALH_CONFIG(ALH_CONFIG_DATA(ALH, 0x102, 48000, 2)))
+')
 
-# 3 HDMI/DP outputs (ID: 5,6,7)
-DAI_CONFIG(HDA, 0, 5, iDisp1,
+# 3 HDMI/DP outputs
+DAI_CONFIG(HDA, 0, HDMI_BELINK_ID, iDisp1,
```

Comment:
no. please don't redefine numbers for devices. If you don't have the amps, then you need to skip the device, not introduce an offset.

[PCMDeviceList.txt](https://github.com/thesofproject/sof/files/6351788/PCMDeviceList.txt)

Reply:
@plbossart This is the BE link ID, not the PCM ID. As you know, the BE link ID is dynamically assigned. If there is DMIC, it will call INC_ID(be_id, cpu_id, link_id), such as 
 at https://github.com/thesofproject/linux/blob/74d194d9a7ec3fa8ec84087e443feb5095637cc8/sound/soc/intel/boards/sof_sdw.c#L1114. This is the same as EXT_AMP. So we have to adjust the DMIC and HDMI BE link ID based on whether we have DMIC or EXT_AMP.

Reply:
Hmm, I wonder why same is not done for sof-icl-rt711-rt1308-rt715-hdmi.m4 .. it also has conditional DAI definitons, but the BE link IDs are static. But both use same sof_sdw.c logic.

Reply:
@kv2019i we have a quirk that sets the RT715 dailink to a fixed value. 

https://github.com/thesofproject/linux/blob/b7cd5a3826b69fef7ab813233c6583d5e774a9fb/sound/soc/intel/boards/sof_sdw_rt715.c#L37

Reply:
@plbossart In https://github.com/thesofproject/linux/blob/b7cd5a3826b69fef7ab813233c6583d5e774a9fb/sound/soc/intel/boards/sof_sdw.c#L199 , we only add the flag for PCH_DMIC. We didn't add the flag SOF_RT715_DAI_ID_FIX. I will add the flag in another patch. In adlp, without the SOF_RT715_DAI_ID_FIX flag, it still works because the DMIC BE link id just happens to be 4. 
Suppose in most cases, we need the SOF_RT715_DAI_ID_FIX flag. After a second thought, I'm wondering whether it is possible that we set RT715 BE link always in the codec init() function unless we set dmic skipping id fix?

Reply:
And if rt715 is not enabled, the HDMI BE link ID will be changed, not from 5. This will be very complicated, depending on:
1. whether rt711 is used, 
2. whether rt1316 is used
3. how rt1316 is used: whether the capture is enabled (it seems we enabled the feedback path on ADLP RVP); is the "NO_AGGREGATION' enabled in the topology.

Maybe using fixed HDMI BE Link IDs will simplify this problem. But I'm not sure whether there is some regression on the old platforms. What's your opinion, @plbossart @kv2019i ?